src/progress/mod.rs

```diff
@@ -116,6 +116,7 @@ impl Progress {
     pub(crate) fn reset(&mut self, next_idx: u64) {
         self.matched = 0;
         self.next_idx = next_idx;
+        self.committed_index = 0;
```

Comment:
Why reset it?

Reply:
Reset it to 0 so we can know whether `committed_index` is stale.

Reply:
Actually you can never know if `committed_index` is stale, because follower can update committed index without informing leader. The only thing is for sure is follower's commit index >= `committed_index`. And the guarantee holds no matter reseting it or not.

Reply:
Yes, it's true, but if we did not reset it, the value will be much staler, and I think it will be quite strange to see it at the log (if people add it to the log).

Reply:
I think 0 is as strange as other numbers. Even stale commit indexes are informative. For example, applications can still rely on the commit indexes from a frequently re-elected leader without waiting for election to complete and a round of heartbeat.

Reply:
OK, let's keep it since it's still useful for the applications.